### PR TITLE
Add featured flag to projects

### LINKED ISSUE #2598: Featured project
This came up in Slack's design thoughts channel. The featured project on the home page should be managed via the API instead of us updating static content every few weeks. What this would needs is:
- a `featured` boolean
- a query param on the `/projects` endpoint

@beckyrother only wants one featured project at a time. We thought that this could be managed by a toggle on the admin page for a project. Perhaps there could be a validation on projects that only allows a single project be `featured: true` at any time?

### PR DESCRIPTION
**Pull request by marten**
Fixes #2598.

# TODO

- [x] Enforce that only 1 project at a time is featured

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: db/migrate/20180110133833_add_featured_to_projects.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddFeaturedToProjects < ActiveRecord::Migration
+  def change
+    add_column :projects, :featured, :boolean, default: false, null: false, index: true
```
Comment (camallen) [inline db/migrate/20180110133833_add_featured_to_projects.rb]: wouldn't this be better with a nullable type / sparse index vs this?

File: app/models/project.rb [hunk 2]
```diff
@@ -57,12 +57,15 @@ class Project < ActiveRecord::Base
 
   validates_inclusion_of :private, :live, in: [true, false], message: "must be true or false"
 
+  validates :featured, uniqueness: {conditions: -> { featured } }
```
Comment (camallen) [inline app/models/project.rb]: this seems like it's overly complex, couldnt the client control how many are featured with a page_size = 1 and a sort order? Also that way we get multiple featured projects and leave the client to figure out which to display?

Comment (marten) [reply app/models/project.rb]: Yeah, that was my thought too, but @beckyrother was very against supporting multiple featured projects. I'm happy to not provide any API support for it if you want.

Comment (camallen) [reply app/models/project.rb]: right - that is until they do ;) Up to you re leaving it and the implementation is fine.
